core/src/data/value/binary_op/i8.rs

```diff
@@ -52,7 +54,29 @@ impl TryBinaryOperator for i8 {
                     .into()
                 })
                 .map(I8),
-            I64(rhs) => Ok(I64(lhs as i64 + rhs)),
+            I64(rhs) => (lhs as i64)
+                .checked_add(rhs)
+                .ok_or_else(|| {
+                    ValueError::BinaryOperationOverflow {
+                        lhs: I8(lhs),
+                        rhs: I64(rhs),
+                        operator: NumericBinaryOperator::Add,
+                    }
+                    .into()
+                })
+                .map(I64),
```

Comment:
This PR is adding `INT(128)` type, so codes changes which affect existing behavior needs to be done in the different PR.

Reply:
Ok.  I will post another PR for changing math to use checked_add etc.  This will allow us to catch overflows and underflows.  Ive added some tests to check for overflows and i will add them to the new PR as well.